simpleclient/src/main/java/io/prometheus/client/exemplars/ExemplarConfig.java

```diff
@@ -0,0 +1,79 @@
+package io.prometheus.client.exemplars;
+
+/**
+ * Static configuration class for Exemplar behavior.
+ */
+public class ExemplarConfig {
+
+  private static final ExemplarConfig instance = new ExemplarConfig();
+
+  private final ExemplarSampler noopExemplarSampler;
+  private final ExemplarSampler defaultExemplarSampler;
+
+  private volatile HistogramExemplarSampler histogramExemplarSampler;
+  private volatile CounterExemplarSampler counterExemplarSampler;
+
+  private ExemplarConfig() {
+    noopExemplarSampler = new NoopExemplarSampler();
+    defaultExemplarSampler = new Tracer().initExemplarSampler(noopExemplarSampler);
+    counterExemplarSampler = defaultExemplarSampler;
+    histogramExemplarSampler = defaultExemplarSampler;
+  }
+
+  /**
+   * @param counterExemplarSampler will be used by default when creating new {@code Counter} metrics.
+   */
+  public static void setCounterExemplarSampler(CounterExemplarSampler counterExemplarSampler) {
+    if (counterExemplarSampler == null) {
+      throw new NullPointerException();
+    }
+    instance.counterExemplarSampler = counterExemplarSampler;
+  }
+
+  /**
+   * @param histogramExemplarSampler will be used by default when creating new {@code Histogram} metrics.
+   */
+  public static void setHistogramExemplarSampler(HistogramExemplarSampler histogramExemplarSampler) {
+    if (histogramExemplarSampler == null) {
+      throw new NullPointerException();
+    }
+    instance.histogramExemplarSampler = histogramExemplarSampler;
+  }
+
+  /**
+   * Disable exemplars by default. Exemplars can still be enabled for individual metrics in the metric's builder.
```

Comment:
Does this actually work? By the time it is called, static initialisation of metrics will likely already be done.

Reply:
I added the following to `README.md` to clarify this
> Note that `ExemplarConfig` is used when a metric is created, so changing the default has no effect on metrics that have already been created.

If we want API to change the `ExemplarSampler` for existing metrics, we'd need to make access to the `ExemplarSampler` thread safe, which would create unexpected synchronization in multi threaded user applications. I don't see a good way of doing this apart from requiring the user to call `ExemplarConfig` before metrics are created.